### PR TITLE
Added chardet encoding detection for file uploads

### PR DESCRIPTION
**Pull request by emrig**

### CODE REVIEW

File: server/tests/test_uploadfile_view.py [hunk 1]
```diff
@@ -14,8 +15,13 @@ def setUp(self):
         self.wfm = load_and_add_module('uploadfile')
         self.factory = APIRequestFactory()
 
-        self.csv_table = pd.read_csv(mock_csv_path)
-        self.xlsx_table = pd.read_excel(mock_xlsx_path)
+        # Path through chardet encoding detection
+        iobytes = open(mock_csv_path, 'rb')
```
Comment (adamhooper) [inline server/tests/test_uploadfile_view.py]: `open()` without `close()` is a faux-pas. The solution is:

```python
with open(mock_csv_path, 'rb') as iobytes:
    self.csv_table = parse_bytesio(iobytes, 'text/csv', None).dataframe
```

File: server/tests/test_uploadfile_view.py [hunk 2]
```diff
@@ -14,8 +15,13 @@ def setUp(self):
...
+        self.csv_table = parse_bytesio(iobytes, 'text/csv', None).dataframe
+
+        iobytes = open(mock_xlsx_path, 'rb')
```
Comment (adamhooper) [inline server/tests/test_uploadfile_view.py]: ditto

**Comment by adamhooper** [comment]
... whoops I've been adding totally unrelated commits to this branch. I'll do the tweak myself and merge, to save ourselves more confusion.

**Comment by emrig** [comment]
Thanks for making the change! Sorry, was offline last night.

**Pull request merged**